scripts/SREP/describe-nodes/script.sh

```diff
@@ -0,0 +1,189 @@
+      error-out
+      ;;
+    -a|--all)
+      all=true
+      ARG_ARRAY=("${ARG_ARRAY[@]:1}")  # shift past argument
+      ;;
+    -m|--master)
+      master=true
+      ARG_ARRAY=("${ARG_ARRAY[@]:1}")  # shift past argument
+      ;;
+    -i|--infra)
+      infra=true
+      ARG_ARRAY=("${ARG_ARRAY[@]:1}")  # shift past argument
+      ;;
+    -w|--worker)
+      worker=true
+      ARG_ARRAY=("${ARG_ARRAY[@]:1}")  # shift past argument
+      ;;
+    -d|--debug)
+      set -x  # Print expanded commands with arguments
+      ARG_ARRAY=("${ARG_ARRAY[@]:1}")  # shift past argument
+      ;;
+    -n|--node|--nodes)
+      if [ ! -z "$nodes" ]
+      then
+        error-out "Only 1 '--nodes' argument can be provided"
+      fi
+      # Check for unset or empty arguments and values that look like a flag/switch
+      if [ ${#ARG_ARRAY[@]} -le 1 ] \
+        || [ -z ${ARG_ARRAY[1]} ] \
+        || [[ ${ARG_ARRAY[1]} =~ ^-.* ]]
+      then
+        error-out "The --nodes requires a list of nodes"
+      fi
+      # Check for ilegal characters in the list of nodes
+      badchars="$(echo ${ARG_ARRAY[1]} | tr -d '[:alnum:]' | tr -d '-' | tr -d '.' | tr -d ',')"
+      if [ ! -z ${badchars} ]
+      then  
+        error-out "'--nodes' can only contain '.|-|,|<alphanumric>' characters.  These characters '${badchars}' are ileagal"
+      fi
+      nodes=${ARG_ARRAY[1]}
+      ARG_ARRAY=("${ARG_ARRAY[@]:2}")  # shift past argument and value
+      ;;
+    -l|--selector)
+      if [ ! -z "$selector" ]
+      then
+        error-out "Only 1 '--selector' argument can be provided"
+      fi
+      # Check for unset or empty arguments and values that look like a flag/switch
+      if [ ${#ARG_ARRAY[@]} -le 1 ] \
+        || [ -z ${ARG_ARRAY[1]} ] \
+        || [[ ${ARG_ARRAY[1]} =~ ^-.* ]]
+      then
+        error-out "'--selector' must contain a node selector"
+      fi
+      badchars="$(echo ${ARG_ARRAY[1]} | tr -d '[:alnum:]' | tr -d '-' | tr -d '.'  | tr -d '/'  \
+          | tr -d ',' | tr -d ':' | tr -d ' ' | tr -d '(' | tr -d ')' | tr -d '!' | tr -d '=')"
+      if [ ! -z ${badchars} ]
+      then
+        error-out "--selector ilegal caharcters in selector.  These characters '${badchars}' are ileagal"
+      fi
+      selector=${ARG_ARRAY[1]}
+      ARG_ARRAY=("${ARG_ARRAY[@]:2}")  # shift past argument and value
+      ;;
+    -*|--*)
+      error-out "Unknown option ${ARG_ARRAY[0]}"
+      ;;
+    *)
+      error-out "Unknown argument"
+      ;;
+  esac
+done
+
+# If --nodes is supplied use that and ignore all other arguments
+# else if --selector is supplied use that ignore the otheres arguments
+# else if --all is supplied use that ignore the otheres arguments
+# else if any of the --master or --infra or --worker arguments are supplied use them all
+if [ ! -z "$nodes" ]
+then
+  
+elif [ ! -z "$selector" ]
+then
+  nodes=$(list-selector $selector)
+elif [ "$all" = true ]
+then
+  add2nodes $(list-selector "node-role.kubernetes.io/master")
```

Comment:
Surround the argument with quotes. (this may look wrong, ie. nested quotes, but it is valid bash)

```suggestion
  add2nodes "$(list-selector "node-role.kubernetes.io/master")"
```

Reply:
Leavening for the linter along with my other digressions... I assume it will scold me appropriately as well (-: